cli-plugins/apps/how-to-guides.hbs.md

```diff
@@ -0,0 +1,611 @@
+      lastTransitionTime: "2022-11-28T05:16:01Z"
+      name: url
+      preview: |
+        http://fluxcd-source-controller.flux-system.svc.cluster.local./gitrepository/default/rmq-sample-app/73c6311eefbf724fee9ad6f4524fa24ec842ff34.tar.gz
+    - digest: sha256:e7884b071fe1bbb2551d42a171043d061a7591e744705572136e689c2a154b7a
+      lastTransitionTime: "2022-11-28T05:16:01Z"
+      name: revision
+      preview: |
+        HEAD/73c6311eefbf724fee9ad6f4524fa24ec842ff34
+    stampedRef:
+      apiVersion: source.toolkit.fluxcd.io/v1beta1
+      kind: GitRepository
+      name: rmq-sample-app
+      namespace: default
+      resource: gitrepositories.source.toolkit.fluxcd.io
+    templateRef:
+      apiVersion: carto.run/v1alpha1
+      kind: ClusterSourceTemplate
+      name: source-template
+  - conditions:
+    - lastTransitionTime: "2022-11-28T05:13:25Z"
+      message: ""
+      reason: ResourceSubmissionComplete
+      status: "True"
+      type: ResourceSubmitted
+    ...
+    inputs:
+    - name: source-provider
+```
+
+## <a id='subpath-usage'> --sub-path
+
+`--sub-path` flag is used to define which path is used as the root path to apply the workload.
+
+- Git repository
+
+    ```bash
+    tanzu apps workload apply subpathtester --git-repo https://github.com/path-to-repo/my-repo --git-branch main --type web --sub-path my-subpath
+
+    🔎 Create workload:
+        1 + |---
+        2 + |apiVersion: carto.run/v1alpha1
+        3 + |kind: Workload
+        4 + |metadata:
+        5 + |  labels:
+        6 + |    apps.tanzu.vmware.com/workload-type: web
+        7 + |  name: subpathtester
+        8 + |  namespace: default
+        9 + |spec:
+       10 + |  source:
+       11 + |    git:
+       12 + |      ref:
+       13 + |        branch: main
+       14 + |      url: https://github.com/path-to-repo/my-repo
+       15 + |    subPath: my-subpath
+    ❓ Do you want to create this workload? [yN]:
+    ```
+
+- Local path
+  - In the directory of the project you want to create the workload from
+
+      ```bash
+      tanzu apps workload apply my-workload --local-path . -s gcr.io/my-registry/my-workload-image --sub-path subpath_folder
+      ❓ Publish source in "." to "gcr.io/my-registry/my-workload-image"? It may be visible to others who can pull images from that repository Yes
+      Publishing source in "." to "gcr.io/my-registry/my-workload-image"...
+      📥 Published source
+      
+      🔎 Create workload:
+            1 + |---
+            2 + |apiVersion: carto.run/v1alpha1
+            3 + |kind: Workload
+            4 + |metadata:
+            5 + |  name: myworkload
+            6 + |  namespace: default
+            7 + |spec:
+            8 + |  source:
+            9 + |    image: gcr.io/my-registry/my-workload-image:latest@sha256:f28c5fedd0e902800e6df9605ce5e20a8e835df9e87b1a0aa256666ea179fc3f
+           10 + |    subPath: subpath_folder
+      ❓ Do you want to create this workload? [yN]:
+
+      ```
+
```

Comment:
***Note:** in cases where a workload must be created from local source code it's recommended to just set the `--local-path` value to point directly to the directory containing the code to be deployed rather than using `--sub-path` to reduce the total amount of code that must be uploaded to create the workload.